llvm/lib/ProfileData/Coverage/CoverageMapping.cpp

```diff
@@ -223,9 +223,127 @@ Expected<int64_t> CounterMappingContext::evaluate(const Counter &C) const {
   return LastPoppedValue;
 }
 
+MCDCTVIdxBuilder::MCDCTVIdxBuilder(
+    std::function<NodeIDs(bool TellSize)> Fetcher) {
+  // Build Nodes and set up each InCount
+  int MaxID = -1;
+  Nodes.resize(std::get<0>(Fetcher(true)));
+  while (true) {
```

Comment:
I think it's fine for structured bindings to live inside loop headers.

```
int ID = 0;
for (auto [ID1, FalseID1, TrueID1] = Fetcher(false); ID1; ID = ID1 - 1;)
```

Reply:
`Fetcher` is expected inside the loop.

Re: correlation between `ID1` and `ID`, see also #81257

Reply:
I don't use `Fetcher` since I have introduced `NextIDsBuilder`.